File: Study/WebView/MainActivity.java
package com.example.webview;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.KeyEvent;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.webkit.WebViewClient;

public class MainActivity extends AppCompatActivity {

    private WebView webView;
    private String url = "https://www.naver.com"; //https:// 도 입력해야 함!

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        webView = (WebView)findViewById(R.id.webView);
        webView.getSettings().setJavaScriptEnabled(true);
        webView.loadUrl(url);
        webView.setWebChromeClient(new WebChromeClient());
        webView.setWebViewClient(new WebViewClientClass()); //처음에 에러가 뜬다. WebViewClientClass라는 이너 클래스를 작성하여 해결.
    }

    //안드로이드의 뒤로가기 버튼을 눌렀을 때, 웹 뷰상에서 이전 페이지로 넘어가는 기능 구현.
    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {

        //뒤로가기 버튼이 눌렸고, 웹뷰에서 이전페이지로 갈 수 있다면.
        if(( keyCode==KeyEvent.KEYCODE_BACK ) && webView.canGoBack()){
            webView.goBack();
            return true;
        }

        return super.onKeyDown(keyCode, event);
    }

    private class WebViewClientClass extends WebViewClient {
        @Override
        public boolean shouldOverrideUrlLoading(WebView view, String url) { //현재 페이지의 url을 읽어오는 메소드.
            view.loadUrl(url);
            return true;
        }
    }
}

File: Study(Kotlin)/Menu/ContextMenu/MainActivity.kt
package com.example.menusample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.ContextMenu
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    private lateinit var textView : TextView;
    private lateinit var textView2 : TextView;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        textView = findViewById(R.id.textView)
        textView2 = findViewById(R.id.textView2)
        registerForContextMenu(textView) //위젯에 contextMenu를 등록함.
        registerForContextMenu(textView2)
    }

    override fun onCreateContextMenu(  // 컨텍스트 메뉴가 최초로 한번만 호출되는 콜백 메서드
        menu: ContextMenu?, v: View?, menuInfo: ContextMenu.ContextMenuInfo?) { //여기서 view는 롱클릭한 위젯
        super.onCreateContextMenu(menu, v, menuInfo)

        if(v == textView) menuInflater.inflate(R.menu.menu_main, menu)
        if(v == textView2) menuInflater.inflate(R.menu.menu_main_kor,menu)
    }

    override fun onContextItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.menu_yellow -> {
                textView.setBackgroundColor(android.graphics.Color.YELLOW)
                true
            }
            R.id.menu_blue -> {
                textView.setBackgroundColor(android.graphics.Color.BLUE)
                true
            }
            R.id.menu_green -> {
                textView.setBackgroundColor(android.graphics.Color.GREEN)
                true
            }
            R.id.menu_yellow_ko -> {
                textView2.setBackgroundColor(android.graphics.Color.YELLOW)
                true
            }
            R.id.menu_blue_ko -> {
                textView2.setBackgroundColor(android.graphics.Color.BLUE)
                true
            }
            R.id.menu_green_ko -> {
                textView2.setBackgroundColor(android.graphics.Color.GREEN)
                true
            }
            else -> super.onContextItemSelected(item)
        }
    }
}
